internal/libs/autofile/autofile.go

```diff
@@ -83,33 +84,68 @@ func OpenAutoFile(path string) (*AutoFile, error) {
 		}
 	}()
 
-	go af.closeFileRoutine()
+	go af.syncFileRoutine()
 
 	return af, nil
 }
 
 // Close shuts down the closing goroutine, SIGHUP handler and closes the
 // AutoFile.
 func (af *AutoFile) Close() error {
-	af.closeTicker.Stop()
+	af.refreshTicker.Stop()
 	close(af.closeTickerStopc)
 	if af.hupc != nil {
 		close(af.hupc)
 	}
 	return af.closeFile()
 }
 
-func (af *AutoFile) closeFileRoutine() {
+func (af *AutoFile) syncFileRoutine() {
 	for {
 		select {
-		case <-af.closeTicker.C:
-			_ = af.closeFile()
+		case <-af.refreshTicker.C:
+			_ = af.syncFile()
 		case <-af.closeTickerStopc:
 			return
 		}
 	}
 }
 
+func (af *AutoFile) syncFile() (err error) {
+	af.mtx.Lock()
+	defer af.mtx.Unlock()
+
+	if af.file == nil {
+		return nil
+	}
+
+	// Firstly let's try to stat to see if the file is still available.
+	if fi, err := af.file.Stat(); err == nil && !fi.IsDir() {
+		// The file disappeared so remove the reference,
+		// and it shall be reopened at the next write.
+		af.file = nil
+		return nil
+	}
+
+	err = af.file.Sync()
+	if err == nil {
+		return nil
+	}
+
+	// There could be multiple reasons why the .Sync failed,
+	// maybe the file permissions changed?
+	_ = af.file.Close()
+
+	// Otherwise, examine if we actually need to create a new file.
+	if errors.Is(err, os.ErrInvalid) {
+		// It shall be opened for the next write.
+		af.file = nil
+		return nil
+	}
+
+	panic(err)
```

Comment:
I don't think we should add new arbitrary panics.

Reply:
I added it because it'll expose unhandled and rare errors given that the routines that previously invoked .closeFile ignored the error. But sure I'll return the error instead.